plugins/woocommerce/src/StoreApi/Routes/V1/Agentic/CheckoutSessionsUpdate.php

```diff
@@ -0,0 +1,215 @@
+	 *
+	 * @var CartController
+	 */
+	protected $cart_controller;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param SchemaController $schema_controller Schema Controller instance.
+	 * @param AbstractSchema   $schema Schema class instance.
+	 */
+	public function __construct( $schema_controller, $schema ) {
+		parent::__construct( $schema_controller, $schema );
+		$this->order_controller = new OrderController();
+		$this->cart_controller  = new CartController();
+	}
+
+	/**
+	 * Get the path of this REST route.
+	 *
+	 * @return string
+	 */
+	public function get_path() {
+		return self::get_path_regex();
+	}
+
+	/**
+	 * Get the path regex for this REST route.
+	 *
+	 * @return string
+	 */
+	public static function get_path_regex() {
+		return '/checkout_sessions/(?P<checkout_session_id>[a-zA-Z0-9._-]+)';
+	}
+
+	/**
+	 * Get method arguments for this REST route.
+	 *
+	 * @return array An array of endpoints.
+	 */
+	public function get_args() {
+		return [
+			'args'   => [
+				'checkout_session_id' => [
+					'description' => __( 'The checkout session ID (Cart-Token JWT).', 'woocommerce' ),
+					'type'        => 'string',
+				],
+			],
+			[
+				'methods'             => \WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'get_response' ],
+				'permission_callback' => [ $this, 'is_authorized' ],
+				'args'                => $this->get_update_params(),
+			],
+			'schema' => [ $this->schema, 'get_public_item_schema' ],
+		];
+	}
+
+	/**
+	 * Get the parameters for updating a checkout session.
+	 *
+	 * @return array Parameters array.
+	 */
+	protected function get_update_params() {
+		$params = AgenticCheckoutUtils::get_shared_params();
+
+		$params['fulfillment_option_id'] = [
+			'description' => __( 'Selected fulfillment option ID.', 'woocommerce' ),
+			'type'        => 'string',
+		];
+
+		return $params;
+	}
+
+	/**
+	 * Check if the request is authorized.
+	 *
+	 * Checks feature enablement and cart token validity.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 * @return bool|\WP_Error True if authorized, WP_Error otherwise.
+	 */
+	public function is_authorized( \WP_REST_Request $request ) {
+		// Check if feature is enabled using helper.
+		$auth_check = AgenticCheckoutUtils::is_authorized( $request );
+		if ( is_wp_error( $auth_check ) ) {
+			return $auth_check;
+		}
+
+		// Additional check for cart token validity.
+		if ( ! $this->has_cart_token( $request ) ) {
+			return new \WP_Error(
+				'woocommerce_rest_invalid_checkout_session',
+				__( 'Invalid or expired checkout session ID.', 'woocommerce' ),
+				array( 'status' => 404 )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Use the checkout_session_id as Cart-Token, and set the respective values to HTTP header and request.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 * @return bool|null
+	 */
+	protected function has_cart_token( \WP_REST_Request $request ) {
+		$session_id = $request->get_param( 'checkout_session_id' );
```

Comment:
Why are we not using `Cart-Token` that exists already?

Reply:
Similar above https://github.com/woocommerce/woocommerce/pull/61216#discussion_r2406480483, we're using the first generated Cart-Token as the `checkout_session_id`. 

In the request coming from the agent, there will be no Cart-Token header.